pkg/image/registryclient/credentials.go

```diff
@@ -0,0 +1,86 @@
+package registryclient
+
+import (
+	"net/url"
+	"sync"
+
+	"github.com/golang/glog"
+
+	"github.com/docker/distribution/registry/client/auth"
+)
+
+var (
+	NoCredentials auth.CredentialStore = &noopCredentialStore{}
+)
+
+type refreshTokenKey struct {
+	url     string
+	service string
+}
+
+type refreshTokenStore struct {
+	lock  sync.Mutex
+	store map[refreshTokenKey]string
+}
+
+func (s *refreshTokenStore) RefreshToken(url *url.URL, service string) string {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+	return s.store[refreshTokenKey{url: url.String(), service: service}]
+}
+
+func (s *refreshTokenStore) SetRefreshToken(url *url.URL, service string, token string) {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+	if s.store == nil {
+		s.store = make(map[refreshTokenKey]string)
+	}
+	s.store[refreshTokenKey{url: url.String(), service: service}] = token
+}
+
+type noopCredentialStore struct{}
+
+func (s *noopCredentialStore) Basic(url *url.URL) (string, string) {
+	glog.Infof("asked to provide Basic credentials for %s", url)
```

Comment:
While you're at it please remove these debug statements here and in the next few functions.